src/schemes/fluid/pressure_acceleration.jl

```diff
@@ -58,6 +58,14 @@ end
     return -volume_term * pressure_tilde * W_a
 end
 
+# Same as `pressure_acceleration_continuity_density`, but using the minus formulation
+# when pressures are negative to avoid tensile instability.
+# See Lyu et al. (2021). https://doi.org/10.1016/j.apor.2021.102938
+@inline function tensile_instability_correction(m_a, m_b, rho_a, rho_b, p_a, p_b, W_a)
+    p = ifelse(p_a >= 0, p_a + p_b, p_b - p_a)
+    return -m_b * p / (rho_a * rho_b) * W_a
```

Comment:
```suggestion
    p_sum_effective = abs(p_a) + abs(p_b)
    return -m_b * p_sum_effective / (rho_a * rho_b) * W_a
```

Your formulation is not correct I think since p_b can be negative as well.